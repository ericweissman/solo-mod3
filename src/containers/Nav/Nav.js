import React from 'react'
import { NavLink } from 'react-router-dom'

const Nav = () => {
  return (
    <nav>
      <h1>AARPlur</h1>
      <NavLink to='/'> Instructions </NavLink>
      <NavLink to='/glitchHop'> Glitch Hop </NavLink>
      <NavLink to='/futureBass'> Future Bass </NavLink>
      <NavLink to='/deepHouse'> Deep House </NavLink>
      <NavLink to='/trap'> Trap </NavLink>
      <NavLink to='/favorites'> Favorites </NavLink>
    </nav>
  )
}

export default Nav