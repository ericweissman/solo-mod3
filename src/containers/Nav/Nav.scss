nav {
  background-color: $color-main;
  font-family: $primary-font;
  display: flex;
  flex-direction: column;
  align-items: center;
  height: 100%;
  position: fixed;
  z-index: 1;
  h1 {
    padding: 30px 0;
    font-size: 45px;
    color: $color-accent;
    font-family: $title-font;
    text-shadow: 3px 3px 3px black;
  }

  .nav-options {
    
    .instructions-nav {
      animation: pulse 2s linear 0s infinite normal; 
      animation-fill-mode: forwards;
      border-bottom: none;
      margin-bottom: 35px;
      &:hover {
        color: white;
        transition: .9s ease-in;
      }
    }
    
    display: flex;
    flex-direction: column;
    align-items: center;

    a {
    color: $color-accent-light;
    width: 205px;
    font-size: 27px;
    padding: 15px 0;
    border-bottom: 2px solid $color-main-dark;
    text-decoration: none;
    text-align: center;

    }


    a:nth-child(n+2) {
      &:hover {
        background-color: $color-accent-light;
        color: $color-main;
        transition: .15s ease-in;
        border: 1px solid transparent;
      }
    }

    a:nth-child(2) {
      border-top: 2px solid $color-main-dark;
      &:hover {
        border-top: 2px solid transparent;
      }
    }
  }

  @keyframes pulse {
    0% {
      transform: scale(1.0)
    }
    50% {
      transform: scale(1.08)
    }
    100% {
      transform: scale(1.0)
    }
  }
}