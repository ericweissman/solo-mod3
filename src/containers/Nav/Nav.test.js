import React from 'react'
import { shallow } from 'enzyme'
import Nav from '../Nav/Nav'

describe('Nav', () => {
  let wrapper
  it('should match the correct snapshot', () => {
    wrapper = shallow(<Nav />)
    expect(wrapper).toMatchSnapshot()
  })
})