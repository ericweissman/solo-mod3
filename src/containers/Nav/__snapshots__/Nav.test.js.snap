// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav should match the correct snapshot 1`] = `
<nav>
  <h1>
    AARPlur
  </h1>
  <NavLink
    activeClassName="active"
    aria-current="page"
    to="/"
  >
     Instructions 
  </NavLink>
  <NavLink
    activeClassName="active"
    aria-current="page"
    to="/glitchHop"
  >
     Glitch Hop 
  </NavLink>
  <NavLink
    activeClassName="active"
    aria-current="page"
    to="/futureBass"
  >
     Future Bass 
  </NavLink>
  <NavLink
    activeClassName="active"
    aria-current="page"
    to="/deepHouse"
  >
     Deep House 
  </NavLink>
  <NavLink
    activeClassName="active"
    aria-current="page"
    to="/trap"
  >
     Trap 
  </NavLink>
  <NavLink
    activeClassName="active"
    aria-current="page"
    to="/favorites"
  >
     Favorites 
  </NavLink>
</nav>
`;
